vsSolutionBuildEvent/vsSolutionBuildEventPackage.cs

```diff
@@ -312,20 +308,113 @@ private void onLogReceived(object sender, Logger.MessageArgs e)
             }
         }
 
+        private void free()
+        {
+            mainToolCmd.Dispose();
+
+            if(errorList != null) {
+                ((IDisposable)errorList).Dispose();
+            }
+
+            ThreadHelper.JoinableTaskFactory.RunAsync(async delegate
+            {
+                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+                if(spSolutionBM != null && _pdwCookieSolutionBM != 0) {
+                    spSolutionBM.UnadviseUpdateSolutionEvents(_pdwCookieSolutionBM);
+                }
+
+                if(spSolution != null && _pdwCookieSolution != 0) {
+                    spSolution.UnadviseSolutionEvents(_pdwCookieSolution);
+                }
+            });
+        }
+
         /// <summary>
-        /// Handler of showing the status-tool window
+        /// Initialization of the package; this method is called right after the package is sited.
         /// </summary>
-        /// <param name="sender"></param>
-        /// <param name="e"></param>
-        private void _menuPanelCallback(object sender, EventArgs e)
+        /// <param name="cancellationToken">A cancellation token to monitor for initialization cancellation, which can occur when VS is shutting down.</param>
+        /// <param name="progress">A provider for progress updates.</param>
+        protected override async Task InitializeAsync(CancellationToken cancellationToken, IProgress<ServiceProgressData> progress)
         {
-            if(StatusTool == null) {
-                throw new ComponentException("Cannot create UI.StatusToolWindow");
+            // When initialized asynchronously, the current thread may be a background thread at this point.
+            // Do any initialization that requires the UI thread after switching to the UI thread.
+            await this.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+            Trace.WriteLine($"Entering Initialize() of: { ToString() }");
+            try
+            {
+                errorList = new VSTools.ErrorList.Pane(this);
+
+                Log._.Received  -= onLogReceived;
+                Log._.Received  += onLogReceived;
+
+                initAppEvents();
+
+                mainToolCmd = await MainToolCommand.initAsync(this, Event);
+                mainToolCmd.setVisibility(false);
```

Comment:
VS now uses lazy loading for all registered plugins. And seems manager will just ignore an autoloads for our plugin when [vsct](https://github.com/3F/vsSolutionBuildEvent/pull/45/files#diff-1762dbfd6c94960f16251b72368abe84R56)  declares `DefaultInvisible` state for the main item:

```xml
        <CommandFlag>DefaultInvisible</CommandFlag>
        <CommandFlag>DynamicVisibility</CommandFlag>
```

that is, I don't see even entering into ctor when we have no activated items at all.

Reply:
[ Invalid ] Misunderstanding was on `OnAfterOpenSolution(object pUnkReserved, int fNewSolution)`. We need to rewrite this logic because of new InitializeAsync

**However,** an lazy loading still produces the case when build can be started **before activation** of our plugin! But this is a features of the modern platform and seems it's not configurable from plugin side at all. Does anyone know some alternatives?

**However,** the end-user still can affect for priority activation via ToolPane window! Just attach once our status panel and it will activate our plugin for early registering handlers for events like opening solution etc.